### PR TITLE
Check case (_id, display_name) and case individuals(individual_id, display_name, phenotype) on case reupload

### LINKED ISSUE #2723: Check what happens if a case is reuploaded with different important keys
Check case:
- Reupload with different display name
- Reupload with different internal id (what happens to the variants, do they point to the correct case?)
- Reupload with different individual IDs
- reupload withn different individual display names.

Perhaps Ideally avoid all these situations and force operator to remove the previous case.

See also #2721 and #2702

### LINKED ISSUE #2771: Update case name command has a misleading description
This is the syntax to update a case:
```
Usage: scout update case [OPTIONS] [CASE_ID]

  Update a case in the database

Options:
  -n, --case-name TEXT           Add/update the display name of case
  -i, --institute TEXT           Update what institutes that has access to
                                 case
  -c, --collaborator TEXT        Add a collaborator to the case
  --vcf PATH                     path to clinical VCF file to be added
  --vcf-sv PATH                  path to clinical SV VCF file to be added
  --vcf-cancer PATH              path to clinical cancer VCF file to be added
  --vcf-cancer-sv PATH           path to clinical cancer structural VCF file
                                 to be added
  --vcf-research PATH            path to research VCF file to be added
  --vcf-sv-research PATH         path to research VCF with SV variants to be
                                 added
  --vcf-cancer-research PATH     path to research VCF with cancer variants to
                                 be added
  --vcf-cancer-sv-research PATH  path to research VCF with cancer structural
                                 variants to be added
  --reupload-sv                  Remove all SVs and re upload from existing
                                 files
  --rankscore-treshold TEXT      Set a new rank score treshold if desired
  --rankmodel-version TEXT       Update the rank model version
  --sv-rankmodel-version TEXT    Update the SV rank model version
  --help                         Show this message and exit.
```

According to the command's description, `-n, --case-name` makes you update a case name, but this is actually incorrect. Case name and institute are instead used to understand which case should be updated whenever a case ID is not provided with the command. Relative code is here: https://github.com/Clinical-Genomics/scout/blob/6171fff1f4b68b03f690b9155fcae16abe76b5e8/scout/commands/update/case.py#L86

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #2723 and fix #2771, check also #2702

Make sure that when a case is re-uploaded it has:
- Same display_name
- Same individuals (cheked keys: individual_id, display_name, phenotype)

Otherwise raise Error.

**How to test**:
1. Load the demo case into a local instance -> `scout setup demo`
2. Modify the demo config file by changing case display name (family_name in config file) and try to update the case with the command ->  `scout --demo load case scout/demo/643594.config.yaml -u` . It should trigger an error
3. Modify the demo config file by changing one of the samples (individual_id, display_name, phenotype) and try to update the case with the command above. It should also trigger error.

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by CR, pytest

### CODE REVIEW

**Comment by northwestwitch** [comment]
@mikaell I could use a review for this one whenever you have time!

**Comment by mikaell** [comment]
> @mikaell I could use a review for this one whenever you have time!

I'll have a look.

File: scout/adapter/mongo/case.py
Comment (mikaell) [inline scout/adapter/mongo/case.py:576]: Function documentation says a list is returned, but here None is returned.

File: scout/adapter/mongo/case.py
Comment (mikaell) [inline scout/adapter/mongo/case.py:605]: The other integrity errors will print a hint to which parameter doesn't match. Might be a good idea to do it here as well?

File: scout/commands/update/case.py
Comment (mikaell) [inline scout/commands/update/case.py:15]: Are commands that used to work removed? Or is something else going on? (I asking because I wonder if removing commands will destroy backwards compatibility and make us do a mayor)

**Review by mikaell — COMMENTED** [review]
Functionality tested locally and works.

Some comments.

**Review by mikaell — APPROVED** [review]
Looks good!

**Comment by northwestwitch** [comment]
I'd like to merge this PR soon (today) as it will avoid problems like #2723 and #2771 in the future. I think it's safe to merge also if will case the next release to be a major.

**Comment by northwestwitch** [comment]
Tested to load a case with this branch on hasta stage: 🆗 
image

This is the case: https://scout-stage.scilifelab.se/cust000/F0006358-mip7_clone

Let's remove it after we merge since has way too many variants!

File: scout/adapter/mongo/case.py
Comment (dnil) [inline scout/adapter/mongo/case.py:582]: But you are checking _id, not display name here?

Comment (northwestwitch) [reply scout/adapter/mongo/case.py:582]: Check later, at line 590

File: scout/adapter/mongo/case.py
Comment (dnil) [inline scout/adapter/mongo/case.py:596]: Why is it important to conserve the phenotype here?

Comment (northwestwitch) [reply scout/adapter/mongo/case.py:596]: I was reasoning that if case individuals change affected status it doesn't have any sense to keep old info, since it means that the analysis has been rerunned, which is like having an entire new case

File: scout/adapter/mongo/case.py
Comment (dnil) [reply scout/adapter/mongo/case.py:582]: Ah, or rather line 655! How about printing the _ids as well: that might be handy. That is just QoL though.😊  
I would also rather drop the 579 comment, or explicitly state that it is assumed that the only other way the _ids should differ is If the display name already existed.

Comment (dnil) [reply scout/adapter/mongo/case.py]: Thanks! 😊

**Comment by northwestwitch** [comment]
I don't understand this new error, I'm not even touching that code! 🤔 

```
images = [{'description': 'A very good description', 'height': 100, 'path': PosixPath('scout/demo/images/custom_images/640x480_... 'scout/demo/images/custom_images/640x480_{REPID}.svg', 'str_repid': '{REPID}', 'title': 'Image for replicon {REPID}'}]

    def _parse_images_cnf(images):
        """
        Parse image info from configuration file and process wildcard searches.
    
        Args:
            images: array with image configuration
    
        Returns:
            parsed_images: array with image binary blob
        """
        parsed_images = []
        for image in images:
            img = image["path"]
>           if "{" in img and "}" in img:
E           TypeError: argument of type 'PosixPath' is not iterable

/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/scout/parse/case.py:59: TypeError
```

**Comment by northwestwitch** [comment]
@dnil  I realized that we'll never going to merge this one if it's a major, so I've reintroduced the cli params and made it a minor. I fixed some stuff also in functions not related to this PR because: **a)** codefactor was complaining for some changes introduced after merging last PR in master, **b)** a test parsing the case config params was no longer working (see PosixPath in comment above). I haven't touched the case and individuals check functions introduced by this before.

**Comment by dnil** [comment]
👍 Do you want a second look at it or are you reasonably confident it is all the same as before?

File: scout/commands/update/case.py
Comment (dnil) [inline scout/commands/update/case.py:16]: 👍

**Comment by northwestwitch** [comment]
> 👍 Do you want a second look at it or are you reasonably confident it is all the same as before?

It should be fine and this time we have to test a re-upload anyway before creating a new release I think..

File: tests/commands/update/test_update_case_cmd.py
Comment (dnil) [reply tests/commands/update/test_update_case_cmd.py:17]: Nice with the tests as well!

**Pull request merged**
